src/core/Track.cpp

```diff
@@ -1148,6 +1136,26 @@ void TrackContentObjectView::mouseReleaseEvent( QMouseEvent * me )
  */
 void TrackContentObjectView::contextMenuEvent( QContextMenuEvent * cme )
 {
+	// Depending on whether we right-clicked a selection or an individual TCO we will have
+	// different labels for the actions.
+	// Get a list of selected selectableObjects
+	QVector<selectableObject *> sos = gui->songEditor()->m_editor->selectedObjects();
+
+	// Convert to a list of selected TCOVs
+	QVector<TrackContentObjectView *> selection;
+	selection.reserve( sos.size() );
+	for( auto so: sos )
+	{
+		TrackContentObjectView *tcov = dynamic_cast<TrackContentObjectView *> ( so );
+		if( tcov != nullptr )
+		{
+			selection.append( tcov );
+		}
+	}
+
+	// Individual TCO or selection being right-clicked?
+	bool individualTCO = selection.contains( this ) ? false : true;
```

Comment:
First of all, you don't need a ternary operator here, you can just use `!selection.contains(this)`.

Secondly, once the duplicate code is extracted, you can write this instead:
`bool individualTCO = getClickedTCOs().size <= 1;`
(Replacing `getClickedTCOs` with whatever we name the new method.)